policyengine_us/variables/gov/states/de/tax/income/deductions/de_itemized_deductions.py

```diff
@@ -0,0 +1,41 @@
+from policyengine_us.model_api import *
+
+
+class de_itemized_deductions(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Delaware itemized deductions"
+    unit = USD
+    definition_period = YEAR
+    reference = (
+        "https://revenuefiles.delaware.gov/2022/PIT-RES_TY22_2022-02_Instructions.pdf#page=7"
+        "https://revenuefiles.delaware.gov/2022/PIT-RES_TY22_2022-02_Instructions.pdf#page=11"
+        "https://casetext.com/statute/delaware-code/title-30-state-taxes/part-ii-income-inheritance-and-estate-taxes/chapter-11-personal-income-tax/subchapter-ii-resident-individuals/section-1109-itemized-deductions-for-application-of-this-section-see-66-del-laws-c-86-section-8"
+    )
+    defined_for = StateCode.DE
+
+    def formula(tax_unit, period, parameters):
+        # 2021 Form M1 instructions say:
+        #   You may claim the Delaware standard deduction or itemize
+        #   your deductions on your Delaware return. You will generally
+        #   pay less Delaware income tax if you take the larger of your
+        #   itemized or standard deduction.
+        # ... calculate pre-limitation itemized deductions
+        itm_deds_less_salt = tax_unit("itemized_deductions_less_salt", period)
```

Comment:
Write your code based on this [form](https://revenuefiles.delaware.gov/2021/TY21_PIT-RSA_2021-01_PaperInteractive.pdf), we already have those variables in the policyengine model, you can directly use them if the calculations are the same as the [federal's](https://www.irs.gov/pub/irs-pdf/f1040sa.pdf).